gammapy/modeling/models/prior.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import numpy as np
+import astropy.units as u
+from gammapy.modeling import PriorParameter, PriorParameters
+from .core import ModelBase
+
+log = logging.getLogger(__name__)
+
+
+def _build_priorparameters_from_dict(data, default_parameters):
+    """Build Parameters object from input dict and default  parameter values."""
+    par_data = []
+
+    input_names = [_["name"] for _ in data]
+
+    for par in default_parameters:
+        par_dict = par.to_dict()
+        try:
+            index = input_names.index(par_dict["name"])
+            par_dict.update(data[index])
+        except ValueError:
+            log.warning(
+                f"PriorParameter '{par_dict['name']}' not defined in YAML file."
+                f" Using default value: {par_dict['value']} {par_dict['unit']}"
+            )
+        par_data.append(par_dict)
+
+    return PriorParameters.from_dict(par_data)
+
+
+class PriorModel(ModelBase):
+    _weight = 1
+    _unit = ""
+
+    def __init__(self, **kwargs):
+        # Copy default parameters from the class to the instance
+        default_parameters = self.default_parameters.copy()
+
+        for par in default_parameters:
+            value = kwargs.get(par.name, par)
+            if not isinstance(value, PriorParameter):
+                par.quantity = u.Quantity(value)
+            else:
+                par = value
+
+            setattr(self, par.name, par)
+
+    @property
+    def parameters(self):
+        """Parameters (`~gammapy.modeling.Parameters`)"""
+        return PriorParameters(
+            [getattr(self, name) for name in self.default_parameters.names]
+        )
+
+    def __init_subclass__(cls, **kwargs):
+        # Add parameters list on the model sub-class (not instances)
+        cls.default_parameters = PriorParameters(
+            [_ for _ in cls.__dict__.values() if isinstance(_, PriorParameter)]
+        )
+
+    @property
+    def weight(self):
+        return self._weight
+
+    @weight.setter
+    def weight(self, value):
+        self._weight = value
+
+    def __call__(self, value):
+        """Call evaluate method"""
+        # assuming the same unit as the PriorParamter here
+        kwargs = {par.name: par.value for par in self.parameters}
+        return self.weight * self.evaluate(value.value, **kwargs)
+
+    def to_dict(self, full_output=False):
+        """Create dict for YAML serialisation"""
+        tag = self.tag[0] if isinstance(self.tag, list) else self.tag
+        params = self.parameters.to_dict()
+
+        if not full_output:
```

Comment:
What happens when the full_output is asked? 
If it is not yet implemented, one should raise a nonImplemented exception

Reply:
Both `full_output = True` and `full_output = False` are implemented and work. If the full output is asked the "min",                     "max" and "error" are set in the dict.